### PR TITLE
Support Qwen3.5/3.6-MoE VL checkpoints in PyTorch-side quantization

### PR DESCRIPTION
**Pull request by titaiwangms**
## Summary

Enables Olive's PyTorch-side quantization passes (`Rtn`/`GPTQ`/`KQuant`, via `ModelWrapper`/`iter_quant_targets`) to work on a Qwen3.5/3.6-MoE checkpoint loaded as its **vision-language** class (`Qwen3_5MoeForConditionalGeneration`, `task="image-text-to-text"`), instead of only the text-only decoder class (`Qwen3_5MoeForCausalLM`) that was previously supported.

### Root cause

Two structural gaps, both purely data/mapping issues (no PyTorch quantization logic needed to change):

1. **Composite config resolution**: the VL checkpoint's top-level `config` has no flat `hidden_size`/`num_hidden_layers`/`num_attention_heads`/`num_key_value_heads`/`head_dim` — they live under `config.text_config`. `ModelWrapper.__init__` resolved these to `None` and crashed with a `TypeError`.
2. **Decoder module path**: the VL decoder is nested at `model.language_model.*` (alongside `model.visual`, the vision tower), not `model.*` directly, so `LAYERS`/`EMBEDDINGS`/`PRE_HEAD_LAYERNORM`/`ROTARY_EMBEDDING` couldn't resolve it.

### Fix

- `olive/assets/io_configs/defaults.yaml`: add `text_config.*` fallbacks for `num_layers`, `hidden_size`, `num_attention_heads`, `num_kv_heads` (mirrors the existing `num_experts` nested-fallback pattern), plus a new `head_dim` alias.
- `olive/common/hf/wrapper.py`: add a `"qwen3_5_moe"` entry to `LAYERS`/`EMBEDDINGS`/`PRE_HEAD_LAYERNORM`/`ROTARY_EMBEDDING` pointing at `model.language_model.*`. Scoped to the VL `model_type` only — the text-only checkpoint carries `model_type == "qwen3_5_moe_text"` and its flat config already matches `"default"`, so it is unaffected (see `test_hf_wrapper_text_only_config_unaffected_by_vl_aliases`).

### Two pre-existing (non-VL-specific) quantization gaps, found while validating the above

Both reproduce identically on the text-only checkpoint too, but affect Qwen3.5/3.6-MoE quantization quality generally:

- `MAMBA` mapping didn't recognize this architecture's `linear_attn` (GatedDeltaNet) attribute name, so its projections (`in_proj_qkv`/`in_proj_a`/`in_proj_b`/`in_proj_z`/`out_proj`) were swept into the generic 2D quantization walk instead of staying full precision like other Mamba/SSM blocks.
- Added a `SHARED_EXPERT_GATE` mapping + `LayerWrapper.get_shared_expert_gate()` accessor (also applicable to `qwen2_moe`/`qwen3_next`/`qwen3_omni_moe`, which use the same attribute name) and wired it into `iter_quant_targets`'s exclusion set — the single-row shared-expert sigmoid gate was being quantized like an ordinary `Linear`, despite being a routing-like signal (same reasoning as excluding the main router).

## Verification

- Confirmed against the real `Qwen/Qwen3.6-35B-A3B` checkpoint's config: alias/mapping resolution (`hidden=2048 heads=16 kv=2 head_dim=256 layers=40`, decoder path `model.language_model.layers`) now matches expectations exactly.
- Built a shape-faithful synthetic VL checkpoint (206M params, real vision tower + 4-layer MoE decoder) and ran the actual `Rtn` pass end-to-end (`bits=4, group_size=32, moe=True, modules_to_not_convert=["visual"]`):
  - `mlp.experts.*` (fused 3D MoE weights) — quantized ✅
  - `mlp.gate` (router), `mlp.shared_expert_gate`, `linear_attn.*` — full precision, untouched ✅
  - `model.visual.*` (vision tower) — 0 quant artifacts ✅
  - Reload of the quantized checkpoint via `HfModelHandler` succeeds.
- Added 5 new unit tests covering composite-config resolution, text-only-path non-regression, and `linear_attn`/`shared_expert_gate` exclusion.
- Full relevant suite: `test/common/`, `test/passes/pytorch/test_rtn.py`, `test/model/` — 622+ passed, 0 regressions (pre-existing unrelated `azure` module errors only). `lintrunner` clean (only the repo-wide `CPY001` false positive).

## Files

- `olive/assets/io_configs/defaults.yaml`
- `olive/common/hf/wrapper.py`
- `olive/common/quant/selection.py`
- `test/common/test_hf_wrapper.py`
- `test/common/quant/test_selection.py`

---
Recreated from #2628 (same branch/commit `b7c0cd3`, pushed directly to microsoft/Olive instead of a fork) so CI has access to the `hf_token` secret, which Azure DevOps withholds from fork PR builds.

### CODE REVIEW

File: olive/common/hf/io_config/io_resolver.py [hunk 1]
```diff
@@ -113,10 +113,23 @@ def get_aliases() -> dict[str, list[str]]:
...
+    unambiguously is simply "not resolvable here", and the caller falls back to the next alias.
+    """
+    for part in attr_path.split("."):
+        try:
+            obj = getattr(obj, part, None)
+        except Exception:  # pylint: disable=broad-except
+            return None
         if obj is None:
             return None
     return obj
```
[the remark this answers is not in the document]
Comment (titaiwangms) [reply olive/common/hf/io_config/io_resolver.py:142]: Confirmed, all real. Fixed in 3b89343d:

1. `_get_nested_attr`: verified via repro that `ModelWrapper(model.model_attributes)` (the actual production path in `transformer_optimization.py`/`dataset.py`/`io_config.py`) leaves `text_config` as a plain `dict` (base `PretrainedConfig.from_dict` doesn't reconstruct model-specific nested sub-configs), so the new `text_config.*` aliases silently resolved to `None` for exactly the VL configs this PR targets. Fixed by using `dict.get` for dict-typed nodes alongside the existing `getattr`/exception handling.

2. Agreed this changes the shared selector for every caller. Added a `quantize_vision` flag (default `False`, matching this PR's intended default behavior) mirroring `quantize_moe`'s opt-in/out shape -- threaded through `OliveHfQuantizationConfig` and exposed as a pass config param on RTN/GPTQ/KQuant (`get_quantizer_config`'s `quantize_vision`), so standalone whole-model quantization callers can opt back in.

3. Removed the outer `preprocessor_config.json` existence guard -- `_save_processor`'s own `_copy_missing_files` already skips existing files one by one, so the outer sentinel was only ever hiding genuinely-missing files (e.g. `chat_template.json`) behind an unrelated file's presence.

File: olive/common/quant/selection.py [hunk 2]
```diff
@@ -314,10 +375,20 @@ def iter_quant_targets(
...
+            skip_ids.add(id(sub))
     # Mamba/SSM blocks stay full precision unconditionally -- see :func:`_collect_mamba_modules`.
     for mamba in _collect_mamba_modules(wrapper):
         for sub in mamba.modules():
             skip_ids.add(id(sub))
+    # A composite VL model's vision tower is never a PyTorch-side quantization target -- see
+    # :func:`_collect_vision_towers`.
+    for tower in _collect_vision_towers(model):
+        for sub in tower.modules():
+            skip_ids.add(id(sub))
```
[the remark this answers is not in the document]
Comment (titaiwangms) [reply olive/common/quant/selection.py]: Agreed -- fixed in 3b89343d by adding a `quantize_vision` flag (default `False`, matching this PR's intended default) that mirrors `quantize_moe`'s opt-in/out shape. It's threaded through `OliveHfQuantizationConfig` and exposed as a pass config param on RTN/GPTQ/KQuant (`get_quantizer_config`'s `quantize_vision`), so callers doing standalone whole-model PyTorch-side quantization (no downstream ONNX vision step) can now opt back in instead of being stuck with vision permanently excluded.

**Comment by titaiwangms** [comment]
Re: the two suppressed review comments on `olive/model/handler/mixin/hf.py`:

- `hf.py:129` (preprocessor_config.json completeness sentinel) -- confirmed and fixed in 3b89343d. `_save_processor`'s own `_copy_missing_files` already skips existing files one by one, so gating the whole call on `preprocessor_config.json`'s existence only hid genuinely-missing files (e.g. `chat_template.json`) whenever some other processor file happened to already exist. Removed the outer guard; added a regression test (`test_save_metadata_processor_fills_in_missing_files_without_overwriting_existing`).
- `hf.py:127` (PR scope/description) -- this is a valid process point (the processor-saving behavior + its own test suite is broader than the PR description's "mapping only" framing), but not a code bug. Leaving the description as-is since this behavior is a real dependency of the VL export path being added here, but flagging it here for visibility.

**Review by jambayk — APPROVED** [review]
